### PR TITLE
feat(vpn): set private network on creation by default

### PR DESCRIPTION
**Pull request by blemoineovh**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #10454 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description
When we create a network service, set it as private by default instead of public

### CODE REVIEW

**Review by dectotam — DISMISSED** [review]

**Review by qpavy — DISMISSED** [review]

**Review by darsene — CHANGES_REQUESTED** [review]
Changing order index in array is quite risky as you can forget some updates.
like for example in the template `add.html` (you forgot an update).
A more secure way is to only update `[this.selectedMode] = this.modes;`

**Comment by blemoineovh** [comment]
> @blemoineovh your commit message is wrong replace `ticket #MANAGER-12914` by `ref: MANAGER-12914`

ok updated commit message

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html [hunk 1]
```diff
@@ -559,7 +559,7 @@ <h1 data-translate="pci_projects_project_instances_add_title"></h1>
 
         <oui-message
             data-type="warning"
-            data-ng-if="$ctrl.subnetGateways.length > 0 && !$ctrl.isGatewayLoading && $ctrl.selectedMode.name !== $ctrl.instanceModeEnum[1].mode"
+            data-ng-if="$ctrl.subnetGateways.length > 0 && !$ctrl.isGatewayLoading && $ctrl.selectedMode.name !== MODE_PUBLIC"
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.html:562]: Where this constant come from ?

**Review by qpavy — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html (hunk 1 above)
Comment (blemoineovh) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.html]: yup i missed a good way to add this constant in the template, i removed it

File: packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js:1]: You've taken the feature in the wrong way. No need to change all the structure of the constants, you just have to change the initialization of `this.selectMode`

L.145 `[this.selectedMode] = this.modes;` -> `[,this.selectedMode] = this.modes;`

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js
Comment (blemoineovh) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js:1]: i can do this, but the mockup did show i have to move the block so i thought it should be done in an other way

Comment (blemoineovh) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js:1]: edit: i kept the block order like before, and selected the second block.

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html [hunk 2]
```diff
@@ -559,7 +559,7 @@ <h1 data-translate="pci_projects_project_instances_add_title"></h1>
 
         <oui-message
             data-type="warning"
-            data-ng-if="$ctrl.subnetGateways.length > 0 && !$ctrl.isGatewayLoading && $ctrl.selectedMode.name !== $ctrl.instanceModeEnum[1].mode"
+            data-ng-if="$ctrl.subnetGateways.length > 0 && !$ctrl.isGatewayLoading && $ctrl.selectedMode.name !== 'public_mode'"
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.html:562]: use the constant `MODE_PUBLIC` instead

File: packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js [hunk 3]
```diff
@@ -1078,8 +1082,8 @@ export default class PciInstancesAddController {
   }
 
   displayNetwork(mode, type) {
-    if (type === this.LOCAL_ZONE_REGION) return mode !== 'public_mode';
-    if (type !== this.LOCAL_ZONE_REGION) return mode !== 'local_private_mode';
+    if (type === this.LOCAL_ZONE_REGION) return mode !== MODE_PUBLIC;
+    if (type !== this.LOCAL_ZONE_REGION) return mode !== MODE_LOCAL_PRIVATE;
     return true;
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js:1087]: return true is never call, you can simplify it
```suggestion
    return  (type === this.LOCAL_ZONE_REGION) ?  mode !== MODE_PUBLIC : !this.isLocalPrivateMode();
```

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/instances/add/add.constants.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
 export const BANDWIDTH_OUT = 'bandwidth_instance_out.consumption';
 export const FILTER_PRIVATE_NETWORK_BAREMETAL = 'ovh.baremetal';
-
+export const MODE_PRIVATE = 'private_mode';
+export const MODE_PUBLIC = 'public_mode';
+export const MODE_LOCAL_PRIVATE = 'local_private_mode';
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.constants.js:5]: your constants are in french spelling

```suggestion
export const PRIVATE_MODE = 'private_mode';
export const PUBLIC_MODE = 'public_mode';
export const LOCAL_PRIVATE_MODE = 'local_private_mode';
```

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js [hunk 5]
```diff
@@ -680,8 +683,9 @@ export default class PciInstancesAddController {
     if (
       modelValue &&
       modelValue.subnet &&
-      this.selectedMode.name !== this.instanceModeEnum[1].mode &&
-      !this.isLocalPrivateMode()
+      this.selectedMode.name !== MODE_PRIVATE &&
+      !this.isLocalPrivateMode() &&
+      this.selectedMode.name !== MODE_PRIVATE
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js]: removed redundance

File: packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js (hunk 3 above)
Comment (blemoineovh) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js]: ok simplified all

File: packages/manager/modules/pci/src/projects/project/instances/add/add.constants.js (hunk 4 above)
Comment (blemoineovh) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.constants.js]: constants refactored to more english style done

**Review by darsene — DISMISSED** [review]

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html (hunk 2 above)
Comment (blemoineovh) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.html]: it is now used in the template

**Review by qpavy — DISMISSED** [review]

**Review by anooparveti — CHANGES_REQUESTED** [review]
Can you please resolve the conflicts?

File: packages/manager/modules/pci/src/projects/project/instances/add/add.constants.js [hunk 6]
```diff
@@ -1,10 +1,12 @@
...
+export const PRIVATE_MODE = 'private_mode';
+export const PUBLIC_MODE = 'public_mode';
+export const LOCAL_PRIVATE_MODE = 'local_private_mode';
 export const INSTANCE_MODES_ENUM = [
-  { mode: 'public_mode' },
-  { mode: 'private_mode' },
-  { mode: 'local_private_mode' },
+  { mode: PRIVATE_MODE },
+  { mode: PUBLIC_MODE },
+  { mode: LOCAL_PRIVATE_MODE },
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.constants.js:9]: @Momie Is it ok to inverse the display order of the tiles finally ?

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html [hunk 7]
```diff
@@ -564,7 +560,7 @@ <h1 data-translate="pci_projects_project_instances_add_title"></h1>
 
         <oui-message
             data-type="warning"
-            data-ng-if="$ctrl.subnetGateways.length > 0 && !$ctrl.isGatewayLoading && $ctrl.selectedMode.name !== $ctrl.instanceModeEnum[1].mode"
+            data-ng-if="$ctrl.subnetGateways.length > 0 && !$ctrl.isGatewayLoading && $ctrl.selectedMode.name !== $ctrl.PUBLIC_MODE"
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.html:563]: If we refer to the legacy code, this message should appear if selectedMode is different than PRIVATE_MODE
as  $ctrl.instanceModeEnum[1] was { mode: 'private_mode' }

**Review by qpavy — COMMENTED** [review]
You can only remove translation from FR file and the other files will be automatically remove.

**Comment by qpavy** [comment]
Maybe you can also replace 'public_mode' and 'private_mode' in https://github.com/ovh/manager/blob/9ff3bfe6c0a6e4858605f17dfc459179cdd081a5/packages/manager/modules/pci/src/projects/project/instances/add/add.html#L496 with the constant you just created

**Review by darsene — APPROVED** [review]

**Review by qpavy — APPROVED** [review]

**Pull request merged**